### DISCUSSION TITLE
Am I installing right?

### DISCUSSION THREAD

**Discussion by mageofthesands** [lshqqytiger/stable-diffusion-webui-directml#207] [Q&A]
--olive, --backend directml, and --onnx doesn't work for me.  It says that the arguments aren't recognized.  SDXL doesn't work either.  That gives me a wall of text about about sizes.  I've tried to launch with no arguments, and no extensions, and no dice either.  I've created a new venv, and set torch to... what is it?  1.x or something?  Not 2.   I'm on version:  •  python: 3.10.6  •  torch: 2.0.0+cpu  •  xformers: N/A  •  gradio: 3.29.0  •  checkpoint: 34bdd8d6a9.
Have I not updated the directML version correctly?  I did git checkout master.  
![sdxlfail](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/81931767/a7814091-ab2f-407f-8175-1b47c929bf4f)

**Comment by rocky533** [comment]
disclaimer: i dont use xl yet so i could be just wrong

u dont need olive or onyx i dont think, i am not even sure xl is fully implemented in this ui yet.

1.5 version
command args for making images less vram use without the no half
 --disable-nan-check --medvram  

command args for inpainting 
 --medvram --no-half --disable-nan-check

You might need --no-half for everything in xl... not sure.

**Comment by lshqqytiger** [comment]
Olive optimization for SDXL is not supported. If you want just SDXL, run without `--olive` to use SGM implementation which is added in 1.5.0.

**Comment by mageofthesands** [comment]
Allow me to clarify.  Olive doesn't work.  Onnx doesn't work.  SDXL doesn't work.  Backend DirectML doesn't work.  Four separate things.  

With or without extensions enabled, using the webui-user of 

```
set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS= --autolaunch --medvram --skip-torch-cuda-test --upcast-sampling --sub-quad-q-chunk-size 512 --sub-quad-kv-chunk-size 512 --sub-quad-chunk-threshold 80  --opt-split-attention-v1 --api --theme dark --use-cpu CLIP BLIP interrogate --no-half
set SAFETENSORS_FAST_GPU=1
call webui.bat
```
Makes SDXL produce a wall of text about mismatched size.  I thought that DirectML version didn't support SDXL, until I read #205 .

**Reply by Rizzlord** [reply]
just reinstall clean, and only use "--disable-nan-check --medvram --autolaunch --api --no-half --no-half-vae --upcast-sampling --opt-sub-quad-attention --precision autocast --use-cpu CLIP BLIP interrogate" After first install, then just use any model to create a fast 512x512. then you can download sdxl1.0 use your arguments and make a 832x832 image, it takes pretty long time but use "lowvram" instead of midvram. It will work but its slow, my advice would be... dont use it for sdxl thats our limit at this moment with amd.

**Comment by mageofthesands** [comment]
Doing a clean install did make the other options show up.  How would I upgrade my main install though?
